packages/stats-worker/src/YieldStats.ts

```diff
@@ -0,0 +1,613 @@
+  apy: number
+  dailyVolume: number
+  rewardToken: string
+}
+
+type Pools = {
+  [token: string]: {
+    [chain: string]: PoolData
+  }
+}
+
+type StakingRewards = {
+  [token: string]: {
+    [chain: string]: {
+      [rewardContractAddress: string]: StakingRewardsData
+    }
+  }
+}
+
+type OptimalYield = {
+  [token: string]: {
+    [chain: string]: OptimalYieldData
+  }
+}
+
+type LegacyYieldData = {
+  [token: string]: {
+    [chain: string]: {
+      apr: number
+      apr7Day: number
+      apr30Day: number
+      stakingApr: number
+      dailyVolume: number
+    }
+  }
+}
+
+type YieldData = {
+  pools: Pools
+  stakingRewards: StakingRewards
+  optimalYield: OptimalYield
+}
+
+type YieldDatas = {
+  legacyYieldData: {
+    timestamp: number
+    data: LegacyYieldData
+  }
+  yieldData: {
+    timestamp: number
+    data: YieldData
+  }
+}
+
+type Response = {
+  yieldDatas: YieldDatas
+}
+
+class YieldStats {
+  sdk = new Hop('mainnet')
+
+  constructor () {
+    this.sdk.setChainProviderUrls({
+      ethereum: ethereumRpc,
+      gnosis: gnosisRpc,
+      polygon: polygonRpc,
+      optimism: optimismRpc,
+      arbitrum: arbitrumRpc
+    })
+
+    console.log(
+      'provider urls:',
+      JSON.stringify(this.sdk.getChainProviderUrls())
+    )
+  }
+
+  async getAllYields () {
+    const timestamp = (Date.now() / 1000) | 0
+    const bridges: any = mainnetAddresses.bridges
+    let yieldData: YieldData = this.initializeYieldData(bridges)
+    const promises: Promise<any>[] = []
+    for (let token in bridges) {
+      for (let chain in bridges[token]) {
+        const shouldSkip = this.shouldSkipYields(bridges, chain, token)
+        if (shouldSkip) {
+          continue
+        }
+
+        promises.push(
+          this.getYieldData(token, chain)
+            .then(res => {
+              console.log(`${chain}.${token} got yield data`)
+              yieldData.pools[token][chain] = {
+                apr: res.apr,
+                apy: res.apy,
+                dailyVolume: res.dailyVolume
+              }
+            })
+            .catch(err => console.error(`apr 1 day ${chain} ${token} error:`, err))
+        )
+        promises.push(
+          this.getStakingYieldData(token, chain)
+            .then(res => {
+              console.log(`${chain}.${token} got staking yield data`)
+              for (const stakingYieldData of res) {
+                yieldData.stakingRewards[token][chain][stakingYieldData.stakingRewardsContractAddress] = {
+                  apr: stakingYieldData.apr,
+                  apy: stakingYieldData.apy,
+                  rewardToken: stakingYieldData.rewardToken,
+                  isOptimalStakingContract: stakingYieldData.isOptimalStakingContract
+                }
+              }
+            })
+            .catch(err => console.error(`staking apr 1 ${chain} ${token} error:`, err))
+        )
+      }
+    }
+
+    await Promise.all(promises)
+    this.removeInactiveBridges(yieldData)
```

Comment:
minor suggestion: return a new object inside of having side effects

```suggestion
yieldData = this.normalizeResult(yieldData)

// ...

function normalizeResult(yieldData) {
  yieldData = this.removeInactiveBridges(yieldData)
  yieldData = this.removeInactiveStakingContracts(yieldData)
  yieldData = this.getOptimalYield(yieldData)
  return yieldData
}

// ...

  removeInactiveBridges (yieldData: YieldData) {
    yieldData = Object.assign({}, yieldData)
     ...
  }
````